### PR TITLE
feat(ssr): implement SSR

### PR DESCRIPTION
**Pull request by tensan8**

### CODE REVIEW

File: src/server/serverRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
+import { Request, Response } from 'express'
+import { renderToString } from 'react-dom/server'
+import { StaticRouter } from 'react-router-dom/server'
+import App from '../app/app'
+import { AssetsProvider } from '../app/common/utils/assetsContext'
+import HtmlTemplate from '../app/htmlTemplate'
+
+export enum ErrorCode {
```
Comment (igarramaddhan) [inline src/server/serverRenderer.tsx:9]: better change to `ServerCode`

File: src/app/common/utils/assetsContext.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+interface Props {
+  assetsMap: any
+}
+
+const AssetsContext = React.createContext<ContextType>({ getAsset: (name: string) => '' })
+
+export const AssetsProvider = (props: PropsWithChildren<Props>): JSX.Element => {
+  const getAsset = (name: string): string => {
```
Comment (igarramaddhan) [inline src/app/common/utils/assetsContext.tsx:14]: wrap with `useCallback`

Comment (igarramaddhan) [reply src/app/common/utils/assetsContext.tsx:14]: also, you can define the `name` param to be more strict by using custom type. you can see `card.tsx`

File: package.json [hunk 3]
```diff
@@ -3,36 +3,43 @@
...
+    "webpack-manifest-plugin": "^5.0.0"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "NODE_ENV=production node src/server",
+    "win:start": "SET NODE_ENV=production& node src/server",
+    "build": "rm -rf dist& NODE_ENV=production webpack --config webpack.config.ts",
```
Comment (igarramaddhan) [inline package.json:36]:
```suggestion
    "build": "rm -rf dist && NODE_ENV=production webpack --config webpack.config.ts",
```

File: src/server/serverRenderer.tsx (hunk 1 above)
Comment (tensan8) [reply src/server/serverRenderer.tsx]: Solved eb24911

File: src/app/common/utils/assetsContext.tsx (hunk 2 above)
Comment (tensan8) [reply src/app/common/utils/assetsContext.tsx]: Solved eb24911

**Review by igarramaddhan — APPROVED** [review]

**Pull request merged**